app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/DefaultGnosisSafeRepository.kt

```diff
@@ -304,6 +315,56 @@ class DefaultGnosisSafeRepository @Inject constructor(
             .subscribeOn(Schedulers.io())
             .map { it.map { TransactionStatus(it.id, it.timestamp, false) } }
 
+
+    override fun createOwner(): Single<Pair<Solidity.Address, ByteArray>> {
+        return Single.just(
+            bip39.mnemonicToSeed(bip39.generateMnemonic(languageId = R.id.english))
+        )
+            .map {
+                val hdNode = KeyGenerator.masterNode(ByteString.of(*it))
+                val key = hdNode.derive(KeyGenerator.BIP44_PATH_ETHEREUM).deriveChild(0).keyPair
+                val privateKey = key.privKeyBytes ?: throw IllegalStateException("Private key must not be null")
+                val address = key.address.asBigInteger()
+                Solidity.Address(address) to privateKey
+            }
+            .subscribeOn(Schedulers.io())
+    }
+
+    override fun saveOwner(safeAddress: Solidity.Address, ownerAddress: Solidity.Address, ownerKey: ByteArray) =
+        Completable.fromCallable {
+            safeDao.insertSafeInfo(GnosisSafeInfoDb(safeAddress, ownerAddress, EncryptedByteArray.create(encryptionManager, ownerKey)))
+        }.subscribeOn(Schedulers.io())
+
+
+    override fun loadOwnerAddress(safeAddress: Solidity.Address): Single<Solidity.Address> {
+        return safeDao.loadSafeInfo(safeAddress)
+            .map {
+                it.ownerAddress
+            }
+            .onErrorResumeNext {
+                accountsRepository.loadActiveAccount().map { it.address }
```

Comment:
Should we check that it is an error that the safe doesn'T exist?

Reply:
it should happen only for safes created with a previous app version that don't have separate owner (no entry in safe info table). In this case safe exists but has device address as its owner